arelle/plugin/validate/ROS/PluginValidationDataExtension.py

```diff
@@ -0,0 +1,87 @@
+"""
+See COPYRIGHT.md for copyright information.
+"""
+
+from __future__ import annotations
+import regex as re
+from arelle import ModelDocument
+from dataclasses import dataclass
+from arelle.ModelXbrl import ModelXbrl
+
+from arelle.utils.PluginData import PluginData
+
+taxonomyReferences = [
+    "https://xbrl.frc.org.uk/ireland/FRS-101/2019-01-01/ie-FRS-101-2019-01-01.xsd",
+    "https://xbrl.frc.org.uk/ireland/FRS-101/2022-01-01/ie-FRS-101-2022-01-01.xsd",
+    "https://xbrl.frc.org.uk/ireland/FRS-101/2022-01-01/ie-FRS-101-2023-01-01.xsd",
+    "https://xbrl.frc.org.uk/ireland/FRS-102/2019-01-01/ie-FRS-102-2019-01-01.xsd",
+    "https://xbrl.frc.org.uk/ireland/FRS-102/2022-01-01/ie-FRS-102-2022-01-01.xsd",
+    "https://xbrl.frc.org.uk/ireland/FRS-102/2022-01-01/ie-FRS-102-2023-01-01.xsd",
+    "https://xbrl.frc.org.uk/ireland/IFRS/2019-01-01/ie-IFRS-2019-01-01.xsd",
+    "https://xbrl.frc.org.uk/ireland/IFRS/2022-01-01/ie-IFRS-2022-01-01.xsd",
+    "https://xbrl.frc.org.uk/ireland/IFRS/2022-01-01/ie-IFRS-2023-01-01.xsd",
+]
+
+schemePatterns = {
+    "http://www.revenue.ie/": re.compile(r"^(\d{7}[A-Z]{1,2}|CHY\d{1,5})$"),
+    "http://www.cro.ie/": re.compile(r"^\d{1,6}$")
+}
+
+TRnamespaces = {
+    "http://www.xbrl.org/inlineXBRL/transformation/2010-04-20",
+    "http://www.xbrl.org/inlineXBRL/transformation/2011-07-31",
+    "http://www.xbrl.org/inlineXBRL/transformation/2015-02-26"
+}
+
+mandatoryElements = {
+    "bus": {
+        "EntityCurrentLegalOrRegisteredName",
+        "StartDateForPeriodCoveredByReport",
+        "EndDateForPeriodCoveredByReport"
+    },
+    "uk-bus": {
+        "EntityCurrentLegalOrRegisteredName",
+        "StartDateForPeriodCoveredByReport",
+        "EndDateForPeriodCoveredByReport",
+    },
+    "ie-dpl": {
+        "DPLTurnoverRevenue",
+        "DPLGovernmentGrantIncome",
+        "DPLOtherOperatingIncome",
+        "DPLGrossProfitLoss",
+        "DPLStaffCostsEmployeeBenefitsExpense",
+        "DPLSubcontractorCosts",
+        "DPLProfitLossBeforeTax"
+    },
+    "core": {
+        "Equity",
+    }
+}
+
+
+@dataclass
+class PluginValidationDataExtension(PluginData):
+
+    _filingTypeInformation: dict[str, set[str | int | list[str]] | int] | None = None
+
+    def filingTypeInformation(self, modelXbrl: ModelXbrl) -> dict[str, set[str|int|list[str]]|int]:
```

Comment:
One of the benefits of this "plugin data" functionality is that we can cache values and avoid recalculating. So I think you can have a check here if `self._filingTypeInformation` is not None, just return it.